R/create_prob_distribution.R

```diff
@@ -152,13 +187,13 @@ create_prob_dist <- function(prob_dist,
         call. = FALSE
       )
     } else {
-      prob_dist <- distributional::dist_truncated(
-        prob_dist,
+      prob_distribution <- distributional::dist_truncated(
+        prob_distribution,
         upper = truncation
       )
     }
   }
 
-  # return prob_dist object
-  prob_dist
+  # return prob_distribution object
+  prob_distribution
```

Comment:
Wrapping the probability distribution object in a subclass? We could do this, but it might have complications for `<distribution>` objects from {distributional} as its built on {vctrs} which is a OOP framework I'm less familiar with.

Please open an issue stating the benefits of giving this object a class and I can implement in a follow-up PR.